File: oauth2-server/src/main/java/a/gleb/keycloaktojwt/db/entity/KeycloakAuthorizationSession.java
package a.gleb.keycloaktojwt.db.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

import static jakarta.persistence.GenerationType.AUTO;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "keycloak_authorization_session")
public class KeycloakAuthorizationSession {

    @Id
    @GeneratedValue(strategy = AUTO)
    @Column(name = "id", nullable = false)
    private UUID id;

    @Column(name = "session_state")
    private String sessionState;

    @Column(name = "scopes")
    private String scopes;

    @Column(name = "access_token_expire_in")
    private LocalDateTime accessTokenExpireIn;

    @Column(name = "username")
    private String username;

    @Column(name = "create_at")
    private LocalDateTime createAt;
}

File: oauth2-server/src/main/kotlin/a/gleb/oauth2server/db/entity/KeycloakAuthorizationSession.kt
package a.gleb.oauth2server.db.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType.AUTO
import jakarta.persistence.Id
import java.time.LocalDateTime
import java.util.*

@Entity(name = "keycloak_authorization_session")
data class KeycloakAuthorizationSession(

    @Id
    @GeneratedValue(strategy = AUTO)
    var id: UUID?,

    @Column(name = "session_state")
    var sessionState: String,

    @Column(name = "scopes")
    var scopes: String,

    @Column(name = "access_token_expire_in")
    var accessTokenExpireIn: LocalDateTime,

    @Column(name = "username")
    var username: String,

    @Column(name = "create_at")
    var createdAt: LocalDateTime
)
